src/sugar/submit.ts

```diff
@@ -54,6 +54,7 @@ async function submitSignedTransaction(
   const request: SubmitRequest = {
     command: 'submit',
     tx_blob: signedTxEncoded,
+    fail_hard: true,
```

Comment:
I think this is good default behavior. There are some cases where users may not want to use fail_hard. In that case, would we tell them to do a `submit` request directly? Or, could we offer this as an option?

Reply:
We could add it as an optional param that defaults to false. But I'm inclined to say that if users know enough about rippled to know that they don't want to `fail_hard`, then it's not hard to call autofill/sign yourself and then directly `submit`. Thoughts?

Reply:
I'm not familiar with this code, or the flow of logic, but one significant downside of using `fail_hard` is that it's a blunt tool, and prevents the transaction from processing in the case of "benign" failures - most notably if the transaction would have been queued. Getting queued usually means that the transaction will succeed, just not right away. In most cases, AFAIK, that's going to meet the user's needs.

All that to say, my vote is to only pass `fail_hard: true` only for `AccountDelete` transactions, or if the user specifies it explicitly.